service/controller/resource/resource.go

```diff
@@ -84,7 +84,7 @@ func New(config Config) ([]resource.Interface, error) {
 				},
 				{
 					NameFunc:      key.CAPICertificateName,
-					NamespaceFunc: key.ClusterID,
+					NamespaceFunc: key.CAPICertificateNamespace,
```

Comment:
Does this work? I mean cause for GS clusters the namespace is still the cluster ID, but `CAPICertificateNamespace` returns the namespace where the `Cluster` object resides, which is usually `default`, right? Or am I missing something?

Reply:
For CAPI clusters, the secret (cluster-id-kubeconfig) is now in the namespace of the cluster cr, not the defautl namespace.
I tested it on all providers first :D

Reply:
OK, cool, thanks!